kittylyst/experiment.py

```diff
@@ -147,3 +148,76 @@ def get_engine(self) -> IEngine:
 
     def get_trial(self) -> ITrial:
         return self._trial
+
+
+class ConfigExperiment(IExperiment):
+    def __init__(self, config: dict):
+        self.config = config
+
+    @property
+    def seed(self) -> int:
+        return self.config["common"].get("seed", super().seed)
+
+    @property
+    def name(self) -> str:
+        return self.config["common"].get("name", super().name)
+
+    @property
+    def hparams(self) -> Dict:
+        hparams = self.config["common"].get("hparams")
+        trial_hparams = (
+            trial.params if (trial := self.get_trial()) is not None else None
+        )
+        default = super().name
+        return hparams or trial_hparams or default
+
+    @property
+    def stages(self) -> List[str]:
+        return list(self.config["stages"].keys())
+
+    def get_stage_params(self, stage: str) -> Dict[str, Any]:
+        return {
+            "num_epochs": self.config["stages"][stage]["num_epochs"],
+            "migrate_model_from_previous_stage": self.config["stages"][
+                stage
+            ].get("migrate_model_from_previous_stage", False),
+        }
```

Comment:
what do you think about `return self.config["stages"][stage]["params"]`? Complicated or better?